spec/system/index_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'spec_helper'
+
+describe 'searching for a bus stop by stop id' do
+  let(:user) { create :user }
+  let!(:bus_stop) { create :bus_stop }
+  before :each do
+    when_current_user_is user
+    visit root_url
+  end
+  context 'correct stop id' do
+    it 'redirects to the edit page' do
+      within 'form', text: 'Enter stop ID' do
+        fill_in 'Enter stop ID', with: bus_stop.hastus_id.to_s
+        click_button 'Search'
+      end
+      expect(page).to have_content "Editing #{bus_stop.name}"
+    end
+  end
+   context 'incorrect stop id' do
+     before :each do
+       within 'form', text: 'Enter stop ID' do
+         fill_in 'Enter stop ID', with: -1
+         click_button 'Search'
+       end
+     end
+     it 'stays on the same page' do
+       expect(page.current_url).to end_with root_path
+     end
+     it 'displays a helpful message' do
+       expect(page).to have_selector 'p.notice',
+         text: 'Stop -1 not found'
+     end
+   end
+   context 'without completing stop id' do
+     it 'autofills' do
+     end
+   end
+end
+
+describe 'searching for a bus stop by stop name' do
```

Comment:
Again this search doesn't seem super user friendly.

Reply:
No it doesn't, but that's not what this PR is about